pfio/AbstractServer.F90

```diff
@@ -121,47 +122,57 @@ end subroutine create_remote_win
 
 contains
 
-   subroutine init(this,comm, port_name, profiler_name, with_profiler)
+   subroutine init(this,comm, port_name, profiler_name, with_profiler, rc)
       class (AbstractServer),intent(inout) :: this
       integer, intent(in) :: comm
       character(*), intent(in) :: port_name
       character(*), optional, intent(in) :: profiler_name
       logical, optional, intent(in) :: with_profiler
+      integer, optional, intent(out) :: rc
 
-      integer :: ierror, MyColor
+      integer :: status, MyColor
```

Comment:
`ierror` is the name of the argument in MPI.   `status` does not buy us much unless it somehow allows the use of a better macro, which does not seem to be the case.

Reply:
Do you mean if we don't want to check it,  keep it as ierror ?

Reply:
Either way we must check the return code.

So either leave it as `ierror` and use `_VERIFY(ierror)`

Or call it `status` and use  `__STAT__` macro (akin to `__RC__` macro).

One could debate the merits of introducing an MPI specific macro `_IERR` that does the same thing with `ierror` but that's probably not worth the trouble.

Reply:
changed